### PR TITLE
method setHeight dynamic

### PR DESCRIPTION
**Pull request by ErikaMolino**
with this method it is possible to dynamically change the height of the backdrop

### CODE REVIEW

**Comment by bfaulk96** [comment]
In recent versions of React Native, you get a warning if you don't specify `useNativeDriver`.... You mind adding `useNativeDriver: false` to that?
Other than that, worked well when I tested it in my project

**Comment by drummerjolev** [comment]
+1. This is a super useful feature. @ErikaMolino thanks for working on this!

File: src/index.js [hunk 1]
```diff
@@ -84,6 +84,15 @@ class RBSheet extends Component {
   close(props) {
     this.setModalVisible(false, props);
   }
+  
+  setHeight(newHeight) {
+    const { animatedHeight } = this.state;
+    const { duration } = this.props;
+    Animated.timing(animatedHeight, {
+      toValue: newHeight,
+      duration
```
Comment (bfaulk96) [inline src/index.js:93]: Add `useNativeDriver: false`

**Comment by radoman93** [comment]
To be honest this is super useful.

**Comment by BasitAli** [comment]
FYI, calling  the `open` method on the bottomsheet ref animates the modal to the new height. for e.g.

```js
  const bottomSheet = useRef();
  useEffect(() => {
    if (visible) {
      bottomSheet.current.open();
    } else {
      bottomSheet.current.close();
    }
  }, [bottomSheet, visible, height]);
```

**Comment by anandhakumarpalanisamy** [comment]
> FYI, calling the `open` method on the bottomsheet ref animates the modal to the new height. for e.g.
> 
> ```js
>   const bottomSheet = useRef();
>   useEffect(() => {
>     if (visible) {
>       bottomSheet.current.open();
>     } else {
>       bottomSheet.current.close();
>     }
>   }, [bottomSheet, visible, height]);
> ```

Works like charm :+1: :100:

**Comment by sachiechan** [comment]
Hi! Is there a plan to merge this PR anytime soon?

**Comment by DibyajyotiMishra** [comment]
> ```js
> useEffect(() => {
>     if (visible) {
>       bottomSheet.current.open();
>     } else {
>       bottomSheet.current.close();
>     }
>   }, [bottomSheet, visible, height]);
> ```

Hey, can you please explain me how to make use of it?

**Comment by anandhakumarpalanisamy** [comment]
> > ```js
> > useEffect(() => {
> >     if (visible) {
> >       bottomSheet.current.open();
> >     } else {
> >       bottomSheet.current.close();
> >     }
> >   }, [bottomSheet, visible, height]);
> > ```
> 
> Hey, can you please explain me how to make use of it?

Hi @DibyajyotiMishra , 

 **You need to create a reference for the bottom sheet**

```tsx
 <RBSheet
          ref={(ref) => {
            this.RBSheet = ref;
          }}
          height={this.getBottomSheetHeight(this.state.pickedImage != null)}
          openDuration={250}>
          {this.AddContentViewWrapper()}
  </RBSheet>
```
Now you can use this reference to change the bottomsheet heigh by simple calling  **() => this.RBSheet?.open()** wherever you feel that the bottomsheet height should  be updated . Calling this will dynamically adjust the bottomsheet height.

The example in the above comment is for hooks.
Below is a sample example using class, 
This is not a complete code. I just extracted the minimum code to give you some idea on how to use this.

The below example contains :  

- a bottomsheet with image picker 
- initial height of the bottomsheet will be minimum
- Whenever a image is selected , we display the image and dynamically adjust the height of the bottomsheet.
- Whenever a image is cancelled, we delete the selected image and dynamically reduce the height of the bottomsheet . 

**Example Demo**


https://user-images.githubusercontent.com/30879156/130783396-2347bcdf-bae2-4c4f-9675-439d44038196.mp4






**Sampe code to give you an idea** 

```tsx
import React from 'react';
import RBSheet from 'react-native-raw-bottom-sheet';
import ImagePicker from 'react-native-image-picker';

class Example extends React.Component {

  // Initialise a reference variable for bottomsheet
  RBSheet?: RBSheet | null;
  constructor(props) {
    super(props);
    this.state = {
      pickedImage: null,
    };
  }

  // function to dynamically set the height of the bottomsheet whenever a image selected or cancelled from image picker
  getBottomSheetHeight = (pickedImage: boolean) => {
    return pickedImage
      ? Dimensions.get('window').height * 0.65
      : Dimensions.get('window').height * 0.2;
  };


  resetAttachmentPicHandler = () => {
    this.setState(
      {
        pickedImage: null,
      },
      () => this.RBSheet?.open(),  // calling this will dynamically adjust the bottomsheet height
    );
  };

  pickAttachmentPicImageHandler = () => {
    var imagePickerOptions = {
      title: 'Pick an Image',
      customButtons: [{name: 'deleteImage', title: 'Delete'}],
    };
    ImagePicker.showImagePicker(imagePickerOptions, (res) => {
      if (res.didCancel) {
        console.log('User cancelled!');
      } else if (res.error) {
        console.log('Error', res.error);
      } else if (res.customButton) {
        if (res.customButton === 'deleteImage') {
          this.resetAttachmentPicHandler();
        }
      } else {
        this.setState(
          {
            pickedImage: {uri: 'data:image/jpeg;base64,' + res.data},
          },
          () => this.RBSheet?.open(), // calling this will dynamically adjust the bottomsheet height
        );
      }
    });
  };

  // The content for the bottomsheet
  // should contain view with buttons to pick and display image or cancel a picked image
  
  AddContentViewWrapper = () => {
    // Create your own form here
    
  }

  render() {
    return (
      <View>

        {/* Button to toggle the bottomsheet */}
        <TouchableOpacity
          style={styles.addContentButtonView}
          onPress={() => this.RBSheet?.open()}>
          <Icon name="pencil" size={25} color="white" />
        </TouchableOpacity>

         {/* Bottomsheet with reference*/}
        <RBSheet
          ref={(ref) => {
            this.RBSheet = ref;
          }}
          height={this.getBottomSheetHeight(this.state.pickedImage != null)}
          openDuration={250}>
          {this.AddContentViewWrapper()}
        </RBSheet>

      </View>
    );
  }
}

```

**Pull request merged**